Userland/Libraries/LibCore/File.cpp

```diff
@@ -191,6 +191,20 @@ String File::current_working_directory()
     return cwd_as_string;
 }
 
+String File::absolute_path(String const& path)
+{
+    if (File::exists(path))
+        return File::real_path_for(path);
+
+    if (path.starts_with("/"sv))
+        return LexicalPath::canonicalized_path(path);
+
+    auto working_directory = File::current_working_directory();
```

Comment:
I'm not sure what we should be doing if the cwd is _not_ unveiled, one thing that comes to mind would be to provide applications with an open fd for their current directory on demand, which they can use with `fstatat`/`openat`/etc to avoid leaking the paths...

Reply:
That does sound like a good simple solution. I'm stuck at conceptualising the program that would function this way without designing the problem away.

From my use of this PR, I found the `getcwd` call is only useful prior to the `unveil` call anyway. If you're trying to figure out where you are after you've veiled the filesystem then you've probably made a mistake.